eis_toolkit/transformations/binarize.py

```diff
@@ -0,0 +1,189 @@
+import numpy as np
+import pandas as pd
+import geopandas as gpd
+import rasterio
+from typing import Optional, Tuple, Union, List, Literal
+
+from eis_toolkit.exceptions import InvalidParameterValueException
+from eis_toolkit.transformations import utils
+from eis_toolkit.checks import parameter
+
+
+# Core functions
+def _binarize_core(  # type: ignore[no-any-unimported]
+    data_array: np.ndarray,
+    threshold: Union[int, float] = int,
+    nodata_value: Optional[int | float] = None,
+) -> np.ndarray:
+    
+    out_array = data_array
+    out_array = utils.replace_nan(data_array=out_array, nodata_value=nodata_value, set_nan=True)
+    out_array = np.ma.array(out_array, mask=np.isnan(out_array))
+    
+    out_array[out_array <= threshold] = 0
+    out_array[out_array > threshold] = 1
+    
+    out_array = out_array.data
```

Comment:
These lines provide the main functionality of binarization. The amount of boilerplate related to just handling multiple bands, geodataframes, multiple columns fill the rest of the file. The boilerplate needs to be reduced.